File: chap12/fig12-18.R
library(dplyr)
library(ggplot2)

d <- read.csv(file='input/data-gp-sparse.csv')
N <- nrow(d)
Np <- 61
Xp <- seq(from=0, to=1, len=Np)

fit_MAP <- readRDS('output/result-model12-10.RDS')
fp <- fit_MAP$draws('fp', format='matrix') %>% as.vector()

d_est <- data.frame(X=Xp, Y=fp)
p <- ggplot(data=d_est) +
  theme_bw(base_size=18) +
  geom_line(aes(x=X, y=Y), size=1) +
  geom_point(data=d, aes(x=X, y=Y), color='black', shape=16, size=0.2, alpha=0.3) +
  labs(x='X', y='Y')
ggsave(p, file='output/fig12-18.png', dpi=300, w=4, h=3)

File: chap12/fig12-18.py
import cmdstanpy
import pandas
import numpy as np
from plotnine import ggplot, aes, theme_bw, geom_line, geom_point, labs

d = pandas.read_csv('input/data-gp-sparse.csv')
N = len(d)
Np = 61
Xp = np.linspace(0, 1, num=Np)

fit = cmdstanpy.from_csv('output/result-model12-10')
fp = fit.stan_variable('fp')

d_est = pandas.DataFrame({'X':Xp, 'Y':fp})
p = (ggplot(d_est)
    + theme_bw(base_size=18)
    + geom_line(aes('X', 'Y'), size=0.5)
    + geom_point(d, aes('X', 'Y'), color='black', shape='.', size=0.2, alpha=0.2)
    + labs(x='X', y='Y')
)
p.save(filename='output/fig12-18.py.png', dpi=300, width=4, height=3)
